services/src/main/java/org/keycloak/services/resources/account/SessionResource.java

```diff
@@ -88,33 +89,31 @@ public SessionResource(KeycloakSession session, Auth auth, HttpRequest request)
     @NoCache
     public Collection<DeviceRepresentation> devices() {
```

Comment:
Maybe this could be slightly changed and return stream instead a collection? 
Something like `distinctByKey` in this [example](https://howtodoinjava.com/java8/java-stream-distinct-examples/).

Reply:
We can change it to stream, but I don't see how to make use of something like you mentioned. If I could map the session into devices then I could filter the devices as shown in that example, but then I would no longer have a reference to the session and it is needed to properly create the device representation.